src/storage/mvcc/txn.rs

```diff
@@ -558,8 +562,11 @@ impl<S: Snapshot> MvccTxn<S> {
                             self.collapse_prev_rollback(key.clone())?;
                         }
 
-                        // insert a Rollback to WriteCF when receives Rollback before Prewrite
-                        let write = Write::new(WriteType::Rollback, ts, None);
+                        // Insert a Rollback to Write CF in case that a stale prewrite command
+                        // is received after a cleanup command.
+                        // During cleanup, when there is no lock, it cannot be the primary key
+                        // of a pessimistic transaction, so the rollback needn't be protected.
+                        let write = Write::new_rollback(ts, false);
```

Comment:
I'm in doubt about the correctness to set this non-protected. Rest LGTM.

Reply:
It is ok because if is no `Lock` exist here, then the corresponding transaction must not a pessimistic transaction's primary.

Reply:
I've update the comments here to:
> Pessimistic transactions prewrite successfully only if all its pessimistic locks exist. So collapsing the rollback of a pessimistic lock is safe. After a pessimistic transaction acquires all its locks, it is impossible that neither a lock nor a write record is found. Therefore, we don't need to protect the rollback here.

Do you think it's OK? @MyonKeminta